walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,97 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> meetingTimesReq = new ArrayList<>();
+    meetingTimesReq.add(TimeRange.WHOLE_DAY);
+    List<TimeRange> meetingTimesOpt = new ArrayList<>();
+    meetingTimesOpt.add(TimeRange.WHOLE_DAY);
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Collections.emptyList();
+    }
+
+    // Mark unavailable times unavailable
+    Collection<String> meetingAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    for (Event event : events) {
+      Collection<String> eventAttendees = event.getAttendees();
+
+      for (String eventAttendee : eventAttendees) {
+        // Make sure all required attendees can make meeting
+        for (String meetingAttendee : meetingAttendees) {
```

Comment:
Have a look at the Collections docs. There's ways you can quickly look for overlaps (e.g. using `Collections.disjoint()`). Even without those, there's probable a `contains` method on collections, so you can likely do something like `meetingAttendees.contains(eventAttendee)`. It turns out they're using HashSets under the hood, so you get a performance boost there.

Reply:
+1

Reply:
I've changed the algorithm to utilize the `disjoint()` method.